lib/src/rules/always_specify_types.dart

```diff
@@ -138,17 +166,95 @@ class _Visitor extends SimpleAstVisitor<void> {
     var name = param.name;
     if (name != null && param.type == null && !name.lexeme.isJustUnderscores) {
       if (param.keyword != null) {
-        rule.reportLintForToken(param.keyword);
-      } else {
-        rule.reportLint(param);
+        var keyword = param.keyword!;
```

Comment:
To avoid this `!`, consider moving the keyword local to before the check?

```dart
var keyword = param.keyword;
if (keyword != null) {
  var type = param.declaredElement?.type;

  ...

}
```




```

Reply:
Done, thanks.